dbms/src/Interpreters/sortBlock.cpp

```diff
@@ -77,9 +84,9 @@ struct PartialSortingLessWithCollation
         for (ColumnsWithSortDescriptions::const_iterator it = columns.begin(); it != columns.end(); ++it)
         {
             int res;
-            if (needCollation(it->first, it->second))
+            if (auto column_string_ptr = needCollation(it->first, it->second))
```

Comment:
This is too complex to be located inside inner loop.
It should be moved out of `operator()`.